lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2401,6 +2401,13 @@ public function clear($entityName = null)
                     $this->doDetach($entity, $visited, false);
                 }
             }
+
+            foreach ($this->entityInsertions as $hash => $entity) {
+                if (get_class($entity) != $entityName) {
```

Comment:
@ricohumme no need to comment on those: they disappear from the diff as soon as you push :-)